File: Scala/src/main/java/synapses/jvm/library/ActivationFunction.java
package synapses.jvm.library;

import synapses.Library;
import synapses.model.netElems.Activation;

public class ActivationFunction {

    Activation contents;

    public ActivationFunction(Activation _contents) {
        contents = _contents;
    }

    public static ActivationFunction sigmoid =
            new ActivationFunction(
                    Library.ActivationFunction$
                            .MODULE$
                            .sigmoid()
            );

    public static ActivationFunction identity =
            new ActivationFunction(
                    Library.ActivationFunction$
                            .MODULE$
                            .identity()
            );

    public static ActivationFunction tanh =
            new ActivationFunction(
                    Library.ActivationFunction$
                            .MODULE$
                            .tanh()
            );

    public static ActivationFunction leakyReLU =
            new ActivationFunction(
                    Library.ActivationFunction$
                            .MODULE$
                            .leakyReLU()
            );

}

File: F#/net/ActivationFunction.cs
using static Synapses.Model.NetElems.Activation;

namespace Synapses.net
{
    public class ActivationFunction
    {
        Function contents;

        public ActivationFunction(Function _contents)
        {
            contents = _contents;
        }

        public static ActivationFunction Sigmoid()
        {
            Function _contents = ActivationFunctionModule.sigmoid;
            return new ActivationFunction(_contents);
        }

        public static ActivationFunction Identity()
        {
            Function _contents = ActivationFunctionModule.identity;
            return new ActivationFunction(_contents);
        }

        public static ActivationFunction Tanh()
        {
            Function _contents = ActivationFunctionModule.tanh;
            return new ActivationFunction(_contents);
        }

        public static ActivationFunction LeakyReLu()
        {
            Function _contents = ActivationFunctionModule.leakyReLU;
            return new ActivationFunction(_contents);
        }
    }
}